### DISCUSSION TITLE
binding of space key

### DISCUSSION THREAD

**Discussion by n-doerrer** [Textualize/textual#973] [Q&A] [answered]
The Bindings class appears to not allow a binding for the space character " ", as the strip() gets rid of it.
```
    def bind(
        self,
        keys: str,
        action: str,
        description: str = "",
        show: bool = True,
        key_display: str | None = None,
        allow_forward: bool = True,
    ) -> None:
        all_keys = [key.strip() for key in keys.split(",")]
```
Is that intentional? Is there a way around, to still bind space or do you generally consider this a bad idea (Im thinking about a "toggle" option)?

**Comment by davep** [comment, marked as answer]
Many keys are bound by their name, and space is one of them. So you'd bind `"space"`.

**Reply by n-doerrer** [reply]
Thank you for your answer. I have been checking textual/keys.py for that and realized that you are absolutely right.
I was just using textual version 0.1.18 which was the newest (non-prerelease) on pypi.org up until 2 days ago.
This works with textual 0.2.1
